src/win/process.c

```diff
@@ -377,11 +377,13 @@ static WCHAR* search_path(const WCHAR *file,
   } else {
     dir_end = path;
 
-    /* The file is really only a name; look in cwd first, then scan path */
-    result = path_search_walk_ext(L"", 0,
-                                  file, file_len,
-                                  cwd, cwd_len,
-                                  name_has_ext);
+    if (NeedCurrentDirectoryForExePathW(L"")) {
```

Comment:
Is the return value something that is expected to change over the lifetime of the program? If not, then perhaps it's a good idea to cache it to keep performance predictable:
```c
static uv_once_t need_current_dir_guard = UV_ONCE_INIT;
static BOOL need_current_dir;

static void need_current_dir_init(void) {
  need_current_dir = NeedCurrentDirectoryForExePathW(L"");
}
```
And then here:
```suggestion
    uv_once(&need_current_dir_guard, need_current_dir_init);
    if (need_current_dir) {
```

Reply:
FI think it will not be changed over a lifetime, it needs to be re-checked again once the env variable is changed. So for caching that is reasonable idea I think

Reply:
It is expected to change over the lifetime of the program. Not particularly recommended or common, but there should not be a performance impact to this call, since we already do quite a few similar calls

Reply:
I see, IMO, but if we store it in to cache we can prevent that value changes over the lifetime and also prevent not recommended actions like that.  Since if we use cache that value will be set again if we’re restarting the application again. CMIIW

But If u guys not prefer that,  I will keep the static function on there and remove the uv_once.

wdyt?

Reply:
Edge cases matter so let's handle it like you originally did, by calling NeedCurrentDirectoryForExePath every time.

Reply:
Reverted